dotnet/src/SemanticKernel.Abstractions/AI/Embeddings/EmbeddingGenerationServiceExtensions.cs

```diff
@@ -154,12 +155,12 @@ public EmbeddingGeneratorEmbeddingGenerationService(
             var attrs = new Dictionary<string, object?>();
             this.Attributes = new ReadOnlyDictionary<string, object?>(attrs);
 
-            var metadata = generator.Metadata;
-            if (metadata.ProviderUri is not null)
+            var metadata = (EmbeddingGeneratorMetadata?)generator.GetService(typeof(EmbeddingGeneratorMetadata));
```

Comment:
This can use the generic GetService

Reply:
Oh, it's because the type is already generic so any extension would require specifying three type parameters.